packages/hash/subgraph/src/types/identifier.ts

```diff
@@ -0,0 +1,78 @@
+import {
+  EntityEditionId,
+  EntityIdAndTimestamp,
+  OntologyTypeEditionId,
+  GraphElementEditionId,
+} from "@hashintel/hash-graph-client";
+import { validate } from "uuid";
+
+// ${AccountId}%${EntityUuid}`
+export type EntityId = `${string}%${string}`;
+
+// ISO-formatted datetime string
+export type EntityVersion = string;
+
+export {
+  EntityIdAndTimestamp,
+  EntityEditionId,
+  OntologyTypeEditionId,
+  GraphElementEditionId,
+};
+
+export const isEntityId = (entityId: string): entityId is EntityId => {
+  const [accountId, entityUuid] = entityId.split("%");
+  return (
+    accountId != null &&
+    entityUuid != null &&
```

Comment:
And I personally really dislike `!!` 😁, especially in this case where I'd be slightly concerned about the implications for things like the nil-uuid.

They're also not equivalent here, `!!""` is `false` while `"" != null` is `true`

Reply:
> And I personally really dislike !! 😁, especially in this case where I'd be slightly concerned about the implications for things like the nil-uuid.

Then maybe we should have an internal discussion about it, because we are actively using it throughout the codebase, and it's a common TS practice.

> things like the nil-uuid

What do you mean by a "nil-uuid"?

> They're also not equivalent here, !!"" is false while "" != null is true

That might actually not be a bad thing in this case, otherwise `isEntityId("%")` would return `true`.

Reply:
> We use it for the workspace account.

I see, for my understanding what's the concern about using `!!` here in relation to the nil-uuid?

Reply:
I think the concern is relatively minimal here, as it should be represented as a string. But UUIDs are also valid u128's, and `!!` involves a lot more strange implicit casting which is just begging for JS to be annoying and give a weird edge-case bug. (`0` being `false` is a consideration with the nil-uuid if we represented it as a number for example)